kedro/io/data_catalog.py

```diff
@@ -116,10 +116,7 @@ class _FrozenDatasets:
     def __init__(self, datasets):
         # Non-alphanumeric characters (except underscore) in dataset name
```

Comment:
It says replace non-alphanumeric characters, but `ñ` in `jalapeño` is a non-alphanumeric character. So this needs to be updated.

Reply:
@921kiyo I can (and considered) change it to read "non-word characters", but I'd argue ñ is part of the Spanish alphabet. :)

Reply:
I think it will work for Chinese or Japanese characters (non-alphabet)? :)